dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/field/FieldHandlerStrategyFactory.java

```diff
@@ -0,0 +1,352 @@
+                field.dbColumn().startsWith("float"), this::floatStrategy);
+
+        addMatchingFieldStrategy(field ->
+                field.dbColumn().startsWith("integer"), this::integerStrategy);
+
+        addMatchingFieldStrategy(field ->
+                field.dbColumn().startsWith("system_field"), this::systemFieldStrategy);
+    }
+
+    private static class SingletonHolder {
+        private static final FieldHandlerStrategyFactory INSTANCE = new FieldHandlerStrategyFactory();
+    }
+
+    /**
+     * Get the instance.
+     * @return FieldHandlerStrategyFactory
+     */
+    public static FieldHandlerStrategyFactory getInstance() {
+
+        return FieldHandlerStrategyFactory.SingletonHolder.INSTANCE;
+    } // getInstance.
+
+    /**
+     * Adds a new field strategy based on a type
+     * @param strategy
+     */
+    public void addFieldStrategy (final String fieldType, final FieldHandlerStrategy strategy) {
+
+        if (Objects.nonNull(fieldType) && Objects.nonNull(strategy)) {
+
+            fieldHandlersMap.put(fieldType, strategy);
+        }
+    }
+
+    /**
+     * Adds a new field strategy associated to a starts with text
+     * @param strategy
+     */
+    public void addStartsWithFieldStrategy (final String fieldTypeText, final FieldHandlerStrategy strategy) {
+
+        if (Objects.nonNull(fieldTypeText) && Objects.nonNull(strategy)) {
+
+            addMatchingFieldStrategy(field -> field.dbColumn().startsWith(fieldTypeText), strategy);
+        }
+    }
+
+    /**
+     * Adds a new field strategy associated to a predicate
+     * @param strategy
+     */
+    public void addMatchingFieldStrategy (final Predicate<Field> fieldTypePredicate, final FieldHandlerStrategy strategy) {
+
+        if (Objects.nonNull(fieldTypePredicate) && Objects.nonNull(strategy)) {
+
+            matchingFieldHandlersList.add(Tuple.of(fieldTypePredicate, strategy));
+        }
+    }
+
+    /**
+     * Get the strategy, throws DotContentletStateException if the strategy is not found for a field type
+     * @param field
+     * @return FieldHandlerStrategy
+     */
+    public FieldHandlerStrategy get (final Field field) {
+
+        final String legacyName = LegacyFieldTypes.getLegacyName(field.getClass());
+        if  (fieldHandlersMap.containsKey(legacyName)) {
+            return fieldHandlersMap.get(legacyName);
+        }
+
+        return matchingFieldHandlersList.stream()
+                .filter(tuple -> tuple._1.test(field))
+                .findFirst()
+                .map(Tuple2::_2)
+                .orElseThrow(() -> new DotContentletStateException("Unable to set value : Unknown field type: " + field));
+    }
+
+    private void doNothingStrategy(final Contentlet contentlet, final Field field, final Object value) throws DotContentletStateException {
+
+        Logger.debug(this,
+                "Cannot set contentlet field value on field type" + field.getClass().getName() + " with value: " + value);
```

Comment:
This message seems to be reporting an error situation, but it's not. Let's change it to something else.